workers/unity/Packages/com.improbable.gdk.core/Worker/MessagesToSend.cs

```diff
@@ -0,0 +1,202 @@
+                    if (typeof(IComponentDiffStorage).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var instance = (IComponentDiffStorage) Activator.CreateInstance(type);
+
+                        componentStorageList.Add(instance);
+                        componentIdToComponentStorage.Add(instance.GetComponentId(), instance);
+
+                        typeToComponentStorage.Add(instance.GetUpdateType(), instance);
+                        foreach (var eventType in instance.GetEventTypes())
+                        {
+                            typeToComponentStorage.Add(eventType, instance);
+                        }
+                    }
+
+                    if (typeof(ICommandDiffStorage).IsAssignableFrom(type) && !type.IsAbstract)
+                    {
+                        var instance = (ICommandDiffStorage) Activator.CreateInstance(type);
+
+                        commandStorageList.Add(instance);
+                        if (!componentIdToCommandIdToStorage.TryGetValue(instance.GetComponentId(),
+                            out var commandIdToStorage))
+                        {
+                            commandIdToStorage = new Dictionary<uint, ICommandDiffStorage>();
+                            componentIdToCommandIdToStorage.Add(instance.GetComponentId(), commandIdToStorage);
+                        }
+
+                        commandIdToStorage.Add(instance.GetCommandId(), instance);
+                    }
+                }
+            }
+        }
+
+        public void Clear()
+        {
+            foreach (var storage in componentStorageList)
+            {
+                storage.Clear();
+            }
+
+            foreach (var storage in commandStorageList)
+            {
+                storage.Clear();
+            }
+
+            worldCommandStorage.Clear();
+            authorityLossAcks.Clear();
+        }
+
+        public void AcknowledgeAuthorityLoss(long entityId, uint componentId)
+        {
+            authorityLossAcks.Add(new EntityComponent(entityId, componentId));
+        }
+
+        public void AddComponentUpdate<T>(T update, long entityId)
+            where T : ISpatialComponentUpdate
+        {
+            var storage = GetComponentDiffStorage(typeof(T));
+
+            // Update ID isn't needed so we set it to 0
+            ((IDiffUpdateStorage<T>) storage).AddUpdate(new ComponentUpdateReceived<T>(update, new EntityId(entityId),
+                0));
+        }
+
+        public void AddEvent<T>(T ev, long entityId) where T : IEvent
+        {
+            var storage = GetComponentDiffStorage(typeof(T));
+
+            // Update ID isn't needed so we set it to 0
+            ((IDiffEventStorage<T>) storage).AddEvent(new ComponentEventReceived<T>(ev, new EntityId(entityId),
+                0));
+        }
+
+        // public void AddCommandRequest<T>(T request, uint componentId, uint commandId) where T : IReceivedCommandRequest
```

Comment:
If this is deliberate - put a TODO above it

Reply:
No I just hadn't got around to it. Done now.